File: 필수문항/136_한국이그리울땐서버에접속하지.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        String word = br.readLine();
        StringTokenizer st = new StringTokenizer(word, "*");
        String prefix = st.nextToken();
        String suffix = st.nextToken();

        test: while (N-- > 0) {
            String pattern = br.readLine();
            if (pattern.length() < word.length() - 1) {
                System.out.println("NE");
            } else {
                for (int i = 0; i < prefix.length(); i++) {
                    if (prefix.charAt(i) != pattern.charAt(i)) {
                        System.out.println("NE");
                        continue test;
                    }
                }
                for (int i = pattern.length() - suffix.length(); i < pattern.length(); i++) {
                    if (suffix.charAt(i - pattern.length() + suffix.length()) != pattern.charAt(i)) {
                        System.out.println("NE");
                        continue test;
                    }
                }
                System.out.println("DA");
            }
        }
    }
}

File: 필수문항/136_한국이그리울땐서버에접속하지.py
import sys

input = sys.stdin.readline

N = int(input())
word = input().rstrip()
idx = word.find("*")
prefix = word[:idx]
suffix = word[idx + 1 :]

for _ in range(N):
    pattern = input().rstrip()
    if len(word) - 1 > len(pattern):
        print("NE")
    else:
        if pattern[:idx] == prefix and pattern[len(pattern) - len(suffix) :] == suffix:
            print("DA")
        else:
            print("NE")
